rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -295,24 +305,101 @@ namespace rsx
 						// Actually uses the modulo operator
 						_min_index = 0;
 						_max_index = attrib.frequency - 1;
+
+						if (freq_count != umax && rsx::method_registers.current_draw_clause.command == rsx::draw_command::indexed)
+						{
+							if (std::find(std::begin(frequencies), std::begin(frequencies) + freq_count, attrib.frequency) == std::begin(frequencies) + freq_count)
+							{
+								frequencies[freq_count++] = attrib.frequency;
+							}
+						}
 					}
 					else
 					{
 						// Same as having no modulo
+						freq_count = umax;
 						_max_index = max_index;
 					}
 				}
 				else
 				{
 					// Division operator
+					freq_count = umax;
 					_min_index = std::min(_min_index, first / attrib.frequency);
 					_max_index = std::max<u32>(_max_index, utils::aligned_div(max_index, attrib.frequency));
 				}
 			}
 		}
 
+		while (freq_count > 0 && freq_count != umax)
+		{
+			const u32 index_size = type == rsx::index_array_type::u32 ? 4 : 2;
+
+			// If min cannot get lower and we can access a bit a more memory than required - do it
+			// The alternative would be re-iterating again over all of them
+			if (_min_index == 0)
+			{
+				if (get_location(real_offset_address) == CELL_GCM_LOCATION_LOCAL)
+				{
+					const auto render = rsx::get_current_renderer();
+
+					if (utils::add_saturate<u32>(real_offset_address - rsx::constants::local_mem_base, (_max_index + 1) * attribute_stride) <= render->local_mem_size)
+					{
+						break;
+					}
+				}
+				else if (vm::check_addr(real_offset_address, vm::page_readable, (_max_index + 1) * attribute_stride))
+				{
+					break;
+				}
+			}
+
+			_max_index = 0;
+			_min_index = first;
+
+			// Force aligned indices as realhw
+			const u32 address = (0 - index_size) & get_address(rsx::method_registers.index_array_address(), rsx::method_registers.index_array_location());
+
+			auto re_evaluate = [&] <typename T> (T)
+			{
+				auto ptr = vm::get_super_ptr<T>(address);
+
+				for (u32 _index = first; _index < first + count; _index++)
+				{
+					const T value = ptr[_index];
+
+					for (u32 freq_it = 0; freq_it < freq_count; freq_it++)
+					{
+						const T res = value % frequencies[freq_it];
+
+						if (res > _max_index)
+						{
+							_max_index = value;
+						}
+
+						if (res < _min_index)
+						{
+							_min_index = value;
+						}
+					}
+				}
+			};
+
+			if (index_size == 4)
+			{
+				re_evaluate(u32{});
```

Comment:
weird right?
I stumbled upon the same issue lol

Reply:
Maybe just pass "auto ptr = vm::get_super_ptr<u32>(address);" here and use auto in the lambda.
Then there would be no such useless param.